BEP7.md

```diff
@@ -0,0 +1,268 @@
+
+## 3. Status
+
+This BEP is under drafting.
+
+## 4. Motivation
+
+Design and issue non-fungible asset on the Binance Chain, as one of the basic economic foundations of the blockchain.
+
+## 5. Collection Specification
+
+### 5.1 Collection Properties
+
+- Token Name: Token Name represents the long name of the token - e.g. "BinanceAnnualEvent2019"
+
+- Denom: Denom is the identifier of the newly issued collection.
+
+- Total Supply: Total supply is the total number of issued NFT tokens belong to this collection.
+
+- NFTs: An array contains items in NFT interface(described below)
+
+```go
+// Collection of non fungible tokens
+type Collection struct {
+  Name          string  `json:"name"`
+  Denom        string  `json:"denom,string,omitempty"` // name of the collection;
+  TotalSupply   int64   `json:"totalSupply"`             // total supply of the NFT in this collection;
+  NFTs          NFTs    `json:"nfts"`                    // NFTs that belong to a collection;
+}
+```
+
+### 5.2 Collection Management Operation
+
+#### 5.2.1 Issue collection
+
+Issuing collection is to create a new unique collection which contains non-fungible token as children on Binance Chain. The new collection represents ownership of something new, and can also peg to existing NFT token from any other blockchains.
+
+| **Field**  | **Type**  | **Description**              |
+| :--------- | :-------- | :---------------------------- |
+| Name       | string    | name of the collection |
+| Denom      | string    | The symbol of the collection |
+| TotalSupply| int64     | total supply of the NFT in this collection |
+| NFTs       | []NFT     | A set of NFT token |
+
+```go
+// NewCollection creates a new NFT Collection
+func NewCollection(name string, denom string, totalSupply uint64, nfts NFTs) Collection {
+  return Collection{
+    Name: strings.TrimSpace(name),
+    Denom: strings.TrimSpace(denom),
+    TotalSupply: uint64(totalSupply),
+    NFTs:  NewNFTs([]NFT(nfts)...),
+  }
+}
+```
+
+**Symbol Convention:**
+
+[Symbol][.B]-[Suffix]
+
+Explanations: Suffix is the first 3 bytes of the issue transaction’s hash. It helps to remove the constraint of requiring unique token names. If this token pegs to an existing blockchain, there should be an additional suffix of “.B”.
+
+**Issue Process:**
+
+- Issuer signed an issue transaction and make it broadcasted to one of Binance Chain nodes
+- This Binance Chain node will check this transaction. If there is no error, then this transaction will be broadcasted to other Binance Chain nodes
+- Issue transaction is committed on the blockchain by block proposer
+- Validators will verify the constraints on total supply and symbol and deduct the fee from issuer’s account
+- New token’s symbol is generated based on the transaction hash. It is added to the issuer’s address and token info is saved on the Binance Chain
+
+## 6. Token Specification
+
+### 6.1 Token Properties
+
+> x/nft/types/nft.go brings metadata on-chain
```

Comment:
as you can see in this [implmentation](https://github.com/cosmos/cosmos-sdk/pull/4209/files#diff-52868bd850fc32806c263c5f9de640d6), it brings metadata to cosmos on-chain.

do we want to bring it on-chain?

pros is data is on the blockchain
cons is metadata is not flexible

Reply:
@oopcode suggests an infoURL in the tokenURI() metadata JSON schema encode metadata, and @okwme references cosmosSDK issues #4046 draft implementation to make it compatible with opensea's API use from coinbase NFT struct TokenID is used as a key when storing the NFT so is not contained in the NFT struct itself.

Reply:
@okwnee @oopcode Billy suggests a compromise betweeen onchain and offchain data v2 separate metadata into an interface with nested features move metadata into its own module that can be used to describe any asset on a SDK chain including (fungible tokens) but specifically references NFT's Metadata/TokenURI/URL's